src/mpid/ch4/src/ch4_probe.h

```diff
@@ -127,13 +126,12 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_improbe_safe(int source,
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_IMPROBE_SAFE);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IMPROBE_SAFE);
 
+#ifdef MPIDI_CH4_USE_WORK_QUEUES
     MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
-
     MPIDI_workq_vci_progress_unsafe();
-    mpi_errno = MPIDI_improbe_unsafe(source, tag, comm, context_offset, av, flag, message, status);
-
     MPID_THREAD_CS_EXIT(VCI, MPIDI_VCI(0).lock);
-
+#endif
+    mpi_errno = MPIDI_improbe_unsafe(source, tag, comm, context_offset, av, flag, message, status);
```

Comment:
I've gotten as far as these patches to move locks to the lower layers. It seems now the "safe" functions just call "unsafe" APIs without any protection. I.e. the unsafe functions are actually safe? Either I'm missing something, or the naming is going to need to be cleaned up.

Reply:
Yes, they can be cleaned up. I am thinking doing that separately as that potentially creates more code conflicts for the down streams.